File: solutions/csharp/resistor-color/ResistorColor.cs
public static class ResistorColor
{
    public static int ColorCode(string color) =>
        ColorValues[color];

    public static string[] Colors() => ColorValues.Keys.ToArray();

    private static readonly IReadOnlyDictionary<string, int> ColorValues = new Dictionary<string, int>
    {
        {"black",  0},
        {"brown",  1},
        {"red",    2},
        {"orange", 3},
        {"yellow", 4},
        {"green",  5},
        {"blue",   6},
        {"violet", 7},
        {"grey",   8},
        {"white",  9}
    };

}

File: solutions/java/resistor-color/src/main/java/ResistorColor.java
import java.util.Arrays;
import java.util.stream.Collectors;

class ResistorColor {
    private enum BandColor {
        black, 
        brown, 
        red, 
        orange, 
        yellow, 
        green, 
        blue, 
        violet, 
        grey, 
        white
}
    int colorCode(String color) {
        return BandColor.valueOf(color).ordinal();
    }

    String[] colors() {
        return Arrays.stream(BandColor.values())
            .map(Enum::name)
            .collect(Collectors.toList())
            .toArray(new String[0]);
    }
}
